ssl/ssl_cert.c

```diff
@@ -378,13 +378,13 @@ void ssl_cert_set_cert_cb(CERT *c, int (*cb) (SSL *ssl, void *arg), void *arg)
 }
 
 /*
- * Verify a certificate chain
+ * Verify a certificate chain/raw public key
  * Return codes:
  *  1: Verify success
  *  0: Verify failure or error
  * -1: Retry required
  */
-int ssl_verify_cert_chain(SSL_CONNECTION *s, STACK_OF(X509) *sk)
+static int ssl_verify_internal(SSL_CONNECTION *s, STACK_OF(X509) *sk, EVP_PKEY *rpk)
```

Comment:
This function is under 100 lines, and would perhaps be simpler if cloned rather than generalised. Consider just making a copy that is RPK-specific, pruned of all excess baggage (including Suite-B support, which is now historical). If the team later decides to unify the code that can be done as a refactor, but I think for now clone is the better option.

Reply:
I hate copying code that could be common. Maintenance can be a pain when a fix has to be done in multiple places. I did originally have this as two separate functions.

That being said, this is still WIP; I'm moving slowly.

Reply:
Understood.  My suggestion is to optimise for clarity over avoiding duplication in this case. It is a judgement call in the end. We can resolve this in later reviews, with more eyes on the code, if you're reluctant to make the change now.

Reply:
@mattcaswell Leaving this question as a judgement call for you and other reviewers.

Reply:
I'd prefer to avoid the code duplication, so I'm with @tmshort on this one.